src/averageAges.js

```diff
@@ -15,11 +15,22 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let menFilter = people.filter(person => person.sex === 'm');
+
+  if (century) {
+    menFilter = people.filter(person => person.sex === 'm'
+      && century === Math.ceil(person.died / 100));
+  }
+
+  const ages = menFilter.map(man => man.died - man.born);
+  const agesArr = ages.reduce((sum, age) => sum + age, 0);
+
+  return agesArr / ages.length;
```

Comment:
Create a separate function to calculate average ages. You use the same logic in different places